src/com/google/enterprise/adaptor/prebuilt/RegexFilter.java

```diff
@@ -0,0 +1,299 @@
+   metadata.transform.pipeline=regexFilter
+   metadata.transform.pipeline.regexFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.RegexFilter.create
+   metadata.transform.pipeline.regexFilter.key=NoIndex
+   metadata.transform.pipeline.regexFilter.decision=do-not-index
+   </code></pre>
+ *
+ * <p>Example: drop the content of documents that have a {@code ContentLength}
+ * greater than or equal to 100 megabytes:
+ * <pre><code>
+   metadata.transform.pipeline=regexFilter
+   metadata.transform.pipeline.regexFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.RegexFilter.create
+   metadata.transform.pipeline.regexFilter.key=ContentLength
+   metadata.transform.pipeline.regexFilter.keyset=params
+   metadata.transform.pipeline.regexFilter.pattern=0*[1-9][0-9]{8,}
+   metadata.transform.pipeline.regexFilter.decision=do-not-index-content
+   </code></pre>
+ *
+ * <p>Example: skips documents whose Metadata {@code Classification} property
+ * is neither {@code PUBLIC} nor {@code DECLASSIFIED}:
+ * <pre><code>
+   metadata.transform.pipeline=regexFilter
+   metadata.transform.pipeline.regexFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.RegexFilter.create
+   metadata.transform.pipeline.regexFilter.key=Classification
+   metadata.transform.pipeline.regexFilter.keyset=metadata
+   metadata.transform.pipeline.regexFilter.pattern=(PUBLIC)|(DECLASSIFIED)
+   metadata.transform.pipeline.regexFilter.when=not-found
+   metadata.transform.pipeline.regexFilter.decision=do-not-index
+   </code></pre>
+ */
+public class RegexFilter implements MetadataTransform {
+  /**
+   * Make decision based upon whether the regular expression matches or not.
+   */
+  private static enum When {
+    FOUND("found"),
+    NOT_FOUND("not-found");
+
+    private final String name;
+
+    private When(String name) {
+      this.name = name;
+    }
+
+    public static When from(String val) {
+      if ("not-found".equalsIgnoreCase(val)) {
+        return When.NOT_FOUND;
+      } else {
+        return When.FOUND;
+      }
+    }
+
+    @Override
+    public String toString() {
+      return name;
+    }
+  };
+
+  // TODO (bmj): remove this and use the one in MetadataTransforms.
+  public enum Keyset {
+    METADATA("metadata"),
+    PARAMS("params");
+
+    private final String name;
+
+    private Keyset(String name) {
+      this.name = name;
+    }
+
+    public static Keyset from(String val) {
+      return (val == null) ? METADATA : Keyset.valueOf(val.toUpperCase());
+    }
+
+    @Override
+    public String toString() {
+      return name;
+    }
+  };
+
+  private static final Logger log
+      = Logger.getLogger(RegexFilter.class.getName());
+
+  /** The name of the key (either Metadata key or params key) to match. */
+  private String key;
+
+  /**
+   * If {@code METADATA}, search the metadata for the specified key;
+   * if {@code PARAMS}, search the params for the specified key;
+   */
+  private Keyset keyset = Keyset.METADATA;
```

Comment:
This value is not needed, since the factory method will always set this, and it prevents the use of final.